### PR TITLE
[GLUTEN-1376][CH] feat: support spill settings: max_bytes_before_external_sort and max_bytes_before_external_group_by

### PR DESCRIPTION
**Pull request by taiyang-li**
## What changes were proposed in this pull request?

- support spill settings: max_bytes_before_external_sort and max_bytes_before_external_group_by 
- fix wrong expression mapping of murmur3hash

(Fixes: \#1376)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by taiyang-li** [comment]
@zzcclp Can you review this pr, thanks very much!

File: backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHTransformerApi.scala [hunk 1]
```diff
@@ -103,6 +103,22 @@ class CHTransformerApi extends TransformerApi with Logging {
...
+    if (nativeConfMap.getOrDefault("spark.memory.offHeap.enabled", "false").toBoolean) {
+      val offHeapSize =
+        nativeConfMap.getOrDefault("spark.gluten.memory.offHeap.size.in.bytes", "0").toLong
+      if (offHeapSize > 0) {
+        val maxBytesBeforeExteralSort = offHeapSize * 0.5
+        nativeConfMap.put(
+          settingPrefix + "max_bytes_before_external_sort",
+          maxBytesBeforeExteralSort.toLong.toString)
+
+        val maxBytesBeforeExternalGroupBy = offHeapSize * 0.5
```
Comment (zzcclp) [inline backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHTransformerApi.scala:117]: does it need to reserve some memory for other opertors ?

Comment (taiyang-li) [reply backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHTransformerApi.scala:117]: We can't control how many memory every operator used precisely. I'm just following the suggestion of ClickHouse
```
    M(UInt64, max_bytes_before_external_group_by, 0, "If memory usage during GROUP BY operation is exceeding this threshold in bytes, activate the 'external aggregation' mode (spill data to disk). Recommended value is half of available system memory.", 0) \

    M(UInt64, max_bytes_before_external_sort, 0, "If memory usage during ORDER BY operation is exceeding this threshold in bytes, activate the 'external sorting' mode (spill data to disk). Recommended value is half of available system memory.", 0) \

```

File: cpp-ch/local-engine/Common/CHUtil.h [hunk 2]
```diff
@@ -111,15 +111,21 @@ class BackendInitializerUtil
 
     static std::map<std::string, std::string> getBackendConfMap(const std::string & plan);
 
-    inline static const String CH_BACKEND_CONF_PREFIX = "spark.gluten.sql.columnar.backend.ch";
-    inline static const String CH_RUNTIME_CONF = "runtime_conf";
-    inline static const String CH_RUNTIME_CONF_PREFIX = CH_BACKEND_CONF_PREFIX + "." + CH_RUNTIME_CONF;
-    inline static const String CH_RUNTIME_CONF_FILE = CH_RUNTIME_CONF_PREFIX + ".conf_file";
+    inline static const String CH_BACKEND_PREFIX = "spark.gluten.sql.columnar.backend.ch";
+
+    inline static const String CH_RUNTIME_CONFIG = "runtime_conf";
```
Comment (liuneng1994) [inline cpp-ch/local-engine/Common/CHUtil.h:116]: runtime_config?

Comment (liuneng1994) [inline cpp-ch/local-engine/Common/CHUtil.h:116]: runtime_config?

**Comment by taiyang-li** [comment]
@zhztheplayer  do you have time to review this pr, thanks very much!

**Review by zzcclp — APPROVED** [review]
LGTM

**Pull request merged**